### PR TITLE
HOSTEDCP-1090: Use statically configured haproxy for router

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
Replace the internal router deployment with a statically configured instance of haproxy. Remove any additional privileges required by control plane workloads.

**Which issue(s) this PR fixes**:
Fixes #[HOSTEDCP-1090](https://issues.redhat.com//browse/HOSTEDCP-1090)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [x] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/ingress/router_config.template [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  mode http
+  monitor-uri /haproxy_ready
+
+resolvers coredns
+  nameserver ns1 {{ .NameServerIP }}:53
+
+{{ $ns := .Namespace }}
+{{- range .Backends }}
+backend {{ .Name }}
+  server {{ .Name }} {{ .DestinationService }}.{{ $ns }}.svc.cluster.local:{{.DestinationPort}} check resolvers coredns init-addr last,libc,none
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/ingress/router_config.template:42]: Do we need the ns and .cluster.local?
Couldn't we just use the name?
If so couldn't  this actually be a static hardcoded list?

**Comment by csrwng** [comment]
/hold
Both router and ignition-server-proxy need to move to request serving nodes when using that isolation

**Comment by csrwng** [comment]
/hold cancel
Updated deployments for request serving

**Review by bryan-cox — APPROVED** [review]
/lgtm

**Comment by sjenning** [comment]
already tagged but lgtm as well :+1:

File: control-plane-operator/controllers/hostedcontrolplane/ingress/router.go [hunk 2]
```diff
@@ -40,86 +41,141 @@ func HCPRouterConfig(hcp *hyperv1.HostedControlPlane, setDefaultSecurityContext
...
+			// If the hypershift.openshift.io/hosted-control-plane label is not present,
+			// then it means the route should be fulfilled by the management cluster's router.
+			continue
+		}
+		switch route.Name {
+		case manifests.KubeAPIServerInternalRoute("").Name,
+			manifests.KubeAPIServerExternalPublicRoute("").Name,
+			manifests.KubeAPIServerExternalPrivateRoute("").Name:
+			p.HasKubeAPI = true
+			continue
```
Comment (muraee) [inline control-plane-operator/controllers/hostedcontrolplane/ingress/router.go:108]: who do we need `continue` here?

Comment (muraee) [reply control-plane-operator/controllers/hostedcontrolplane/ingress/router.go:108]: just curious, otherwise lgtm

Comment (csrwng) [reply control-plane-operator/controllers/hostedcontrolplane/ingress/router.go:108]: we don't need it :) 
I had that there before that section was inside the case stmt but forgot to remove. It doesn't hurt, but I will remove it in a follow up

**Comment by csrwng** [comment]
Need to consult with IBM Cloud team to determine whether an override for the PSA labels is needed before backporting
cc @rtheis

**Comment by rtheis** [comment]
@csrwng we ran some tests and confirmed that `baseline` PSA labels will work for us but not `restricted`.  If you all intend to proceed with `restricted` PSA labels then we will need https://github.com/openshift/hypershift/pull/2886 to override the default.

**Pull request merged**